releasenotes/notes/fix-instruction-c_if-3334bc8bcc38a327.yaml

```diff
@@ -0,0 +1,15 @@
+---
+deprecations:
+  - |
+    Creating an instance of :obj:`.InstructionSet` with the ``circuit_cregs`` keyword argument is deprecated.
+    In general, these classes never need to be constructed by users (but are used internally), but should you need to, you should pass a callable as the ``resource_resolver`` keyword argument.
```

Comment:
I think having an example of how this is done would be useful. I agree it's not really user facing, but reading this with my pretend naive user eye I don't know what this would look like.

Reply:
Sounds good, done in 2c608d212.